### PR TITLE
DEX-410 added support and tests for collaboration patching

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Support for collaboration patching and testing of it
- Fixed a bug in the checking of the read access
---

### CODE REVIEW

File: tests/modules/collaborations/resources/test_collaboration_patch.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    flask_app_client, researcher_1, readonly_user, researcher_2, db
+):
+    from app.modules.collaborations.models import Collaboration
+
+    collab = None
+    try:
+        data = {'user_guid': str(researcher_2.guid)}
+        collab_utils.create_collaboration(flask_app_client, researcher_1, data)
+        collabs = Collaboration.query.all()
+        collab = collabs[0]
```
Comment (colinwkingen) [inline tests/modules/collaborations/resources/test_collaboration_patch.py:17]: This won't work when testing with a production DB or something like the soon to be pseudo-migrated zebra database.

File: tests/modules/collaborations/resources/test_collaboration_patch.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        # should work
+        patch_data = [utils.patch_replace_op('title', 'Some sort of title')]
+        collab_utils.patch_collaboration(
+            flask_app_client,
+            collab_guid,
+            researcher_2,
+            patch_data,
+        )
+        assert collab.title == 'Some sort of title'
```
Comment (colinwkingen) [inline tests/modules/collaborations/resources/test_collaboration_patch.py:30]: We probably don't even need title, that's the old boilerplate cruft that got added to all models prior to being cut.

**Review by colinwkingen — APPROVED** [review]
Looks pretty good to me, approving but will wait for a reply on my comments before hitting the button.

File: tests/modules/collaborations/resources/test_collaboration_patch.py (hunk 1 above)
Comment (hwindsor) [reply tests/modules/collaborations/resources/test_collaboration_patch.py:17]: The same applies to a whole ton of tests. I'm afraid that will be a big task when we do that.

File: tests/modules/collaborations/resources/test_collaboration_patch.py (hunk 2 above)
Comment (hwindsor) [reply tests/modules/collaborations/resources/test_collaboration_patch.py]: If it's cruft, it's gone, new push has removed it

**Pull request merged**